maintenance/projects/js-toolkit/packages/liferay-npm-build-tools-common/src/project/index.ts

```diff
@@ -362,18 +362,55 @@ export class Project {
 		}
 	}
 
-	_loadNpmbundlerrc(): void {
-		const npmbundlerrcPath = this._configFile.asNative;
+	_findAutopresets(): string[] {
+		const {dependencies, devDependencies} = this._pkgJson;
+
+		return Object.keys({
+			...(dependencies || {}),
+			...(devDependencies || {}),
+		}).reduce((autoPresets, pkgName) => {
+			try {
+				const {dependencies} = this.require(pkgName + '/package.json');
+
+				if (dependencies && dependencies['liferay-npm-bundler']) {
+					autoPresets.push(pkgName);
+				}
+			}
+			catch (err) {
 
-		const config = fs.existsSync(npmbundlerrcPath)
-			? readJsonSync(npmbundlerrcPath)
-			: {};
+				// ignore
 
-		// Apply preset if necessary
+			}
 
+			return autoPresets;
+		}, []);
+	}
+
+	_loadDefaultPreset(): string {
 		let presetFilePath;
 
-		if (config.preset === undefined) {
+		const autoPresets = this._findAutopresets();
+
+		if (autoPresets.length > 1) {
+			throw new Error(
+				'Multiple autopreset dependencies found in project (' +
+					autoPresets +
+					'): please remove the invalid ones or ' +
+					'explicitly define the preset to be used in the ' +
+					'.npmbundlerrc file'
+			);
+		}
+		else if (autoPresets.length) {
+			presetFilePath = this.resolve(autoPresets[0]);
+
+			this._toolsDir = new FilePath(
```

Comment:
Not sure how much logging we're doing, but it might not be a bad idea to somehow log (here or elsewhere) which preset is being used and if relevant, why. Might save us and customers from some head scratching in the future.

Reply:
All those details are shown in the report file (we decided to dump all relevant build information there long ago, since the logging was unmanageable), but it's true that we don't write the name of the preset being used, just the resultant configuration, so I'll add it too :+1:

Reply:
> [...] we decided to dump all relevant build information there long ago [...]

I dispute that. Especially the "**all**" portion of it. I think we should revisit that decision. It's fine to drop a massive amount information in a report that customers know to activate and navigate or not. I don't think that's contrary to making our output friendlier.

If we're going to make relevant decisions for the users, it wouldn't hurt to let them know in advance. Compare the following workflows after an error:

// Workflow 1
- Check log, see "Compiling with preset liferay-ce-ga1-7.2"
- Oops... I'm building for 7.3
- Update it

// Workflow 2
- Check log, see nothing
- Edit `.npmbundlerrc` to add `report: true`
- Build again to generate a report
- Locate the report.html file and open it
- Scan it to find "Compiling with preset liferay-ce-ga1-7.2"
- Oops... I'm building for 7.3
- Update it

Reply:
I like the idea of having output. I always thought that not logging anything wasn't a good idea, but we decided to remove everything in the past because we didn't want any message to show up in the console.

We also have a [verbose](https://github.com/liferay/liferay-frontend-projects/blob/master/maintenance/projects/js-toolkit/docs/.npmbundlerrc-file-reference.md#verbose) flag but it can only be passed in the `.npmbundlerrc` (and not as an argument, like we can do for the report and the JAR file creation) so it's a bit ~complex to use~ useless.

We may:

1) Dump some output by default setting the `verbose` flag's default value to (now it is `false` by default). This is what webpack does, IINM.
2) Don't show anything and provide a CLI switch to activate verbosity.
3) Other thing...

I would go with 1 since it's the most usual way of working for tools like this...

Reply:
> but we decided to remove everything in the past because we didn't want any message to show up in the console.

Maybe we could slowly move towards logging simple messages? There is definitely a fine line between helpfulness and noise, but I do enjoy small one-liners that let me know something happened.

For example, here is a Next.js server.
![Screen Shot 2021-05-19 at 11 18 12 AM](https://user-images.githubusercontent.com/6843530/118863561-e6fd5f00-b893-11eb-89f9-934aafba8f25.png)

I really like how they have statuses for messages like "Event" and "Wait". Now this is different because its a live server, but I gain confidence as a developer when I see information like that.

In our case here, I like the idea of something as simple as `JS Bundler: Compiling with preset liferay-ce-ga1-7.2` and then we can use the verbose flag to dump more granular details of the build. We could just add a line somewhere in this output 
![Screen Shot 2021-05-19 at 11 23 02 AM](https://user-images.githubusercontent.com/6843530/118864271-ab16c980-b894-11eb-9796-33b4b35795fb.png)

Reply:
> Maybe we could slowly move towards logging simple messages? There is definitely a fine line between helpfulness and noise, but I do enjoy small one-liners that let me know something happened.

That's [already done](https://github.com/izaera/liferay-js-toolkit/commit/02e8806465f90cab886f61f6e0ebab9ae10c2811). The only caveat is that, by default [we set `verbose` to `false`](https://github.com/izaera/liferay-js-toolkit/commit/02e8806465f90cab886f61f6e0ebab9ae10c2811#diff-9235b7a19f38965392ca7ba4ffb6347143a9ebda3b395fc93879f003d788f1baR177) and, to change it, you need to create a .npmbundlerrc file and set it to `true`.

So, my proposal was:

1. Create a `verbose` command line switch as we have for [--dump-report](https://github.com/liferay/liferay-frontend-projects/blob/master/maintenance/projects/js-toolkit/docs/.npmbundlerrc-file-reference.md#--dump-report) (that could be specified both in the command line and in the npmbundlerrc file)
2. Change `verbose`'s default to `true` so that we log some information about what's happening (as we did in the past, before `verbose` was introduced)
3. Add a log message for the preset in use (if any)
4. Refine any other messages we are logging already

I will create a [new issue](https://github.com/liferay/liferay-frontend-projects/issues/552) for this, since it can be done detached from this PR.

Reply:
> use the verbose flag to dump more granular details of the build

That's what the report file does. Due to the complexity of the bundler builds, where many things can go wrong deep inside third party packages when we find weird code, we create a HTML file on user's request to explain what is happening under the hood and what decisions are taken when building the project.

I think it is better to simply:

1. Log essential messages by default as proposed in the issue I've created
2. Let users disable verbosity so that nothing is printed in the console: this is useful for cases like liferay-portal where we don't want to print anything (the reason why the [commit linked above](https://github.com/izaera/liferay-js-toolkit/commit/02e8806465f90cab886f61f6e0ebab9ae10c2811) was merged), or similar projects where the bundler is called multiple times and you don't really care about the output in the full log, only on a per-project basis, when anything goes wrong.
3. Dump any other info to the report file. 
 
The report file is one of the first things [we ask users for](https://github.com/liferay/liferay-frontend-projects/blob/master/maintenance/projects/js-toolkit/docs/How-to-troubleshoot-your-setups.md) when it comes to support them, and it has helped several people to understand a bit better what the bundler does under the hood, so I think it's worth having it.

We could even increase its functionality with things like [this table of detailed explanations about bundler messages](https://github.com/liferay/liferay-frontend-projects/blob/master/maintenance/projects/js-toolkit/docs/Report-messages.md) which the bundler report files link directly.

Reply:
Of course, as @jbalsas says in his workflow 2, essential things should go in the log and report should only be used for things that can break.

However, it is sometimes unclear what essential is, because in that example, we consider the preset essential because it leads to user confusion. But that's only because of the way the case is explained that it's centering the cause of the error in the target platform (something that the user will probably not change so often and not very error prone IMO): 

I mean, what are the chances that you need to change the target platform? And in that case, what are the chances of configuring a bad platform? We are assuming that it will happen often and thus we are showing a message log in the console. But we can apply the same reasoning to anything that could go wrong.

I mean, whenever you encounter an error you wish the tool/framework you are using would notify you clearly about that if it didn't, but most of the times that's not the case :shrug: 

On the other hand, if you clog the console with "useful" log messages that hint you about possible errors (aka: warnings) there comes a time when it stops being useful and it's like having nothing.

So, I lean more toward to what @bryceosterhaus says: 

> I gain confidence as a developer when I see information like that.

So, I propose not to focus on what is needed in the log to avoid problems, but more on what we would like to see as developers to get a feeling of what's going on.

To finish: in this case I would like to see the autopreset in the console because it's my target platform, and I would also like to see the typical finalization message as well as some intermediate messages to know what the tool is doing (for example: one per bundled package or something like that) and know if it stuck, for example.